avocado/core/runners/requirement_asset.py

```diff
@@ -0,0 +1,108 @@
+import time
+from multiprocessing import Process, SimpleQueue
+
+from ...utils import data_structures
+from ...utils.asset import Asset
+from .. import nrunner
+from ..settings import settings
+
+
+class RequirementAssetRunner(nrunner.BaseRunner):
+    """Runner for requirements of type package
+
+    This runner handles the fetch of files using the Avocado Assets utility.
+
+    Runnable attributes usage:
+
+     * kind: 'requirement-asset'
+
+     * uri: not used
+
+     * args: not used
+
+     * kwargs:
+        - name: the file name or uri (required)
+        - asset_hash: hash of the file (optional)
+        - algorithm: hash algorithm (optional)
+        - locations: location(s) where the file can be fetched from (optional)
+        - expire: time in seconds for the asset to expire (optional)
+    """
+
+    @staticmethod
+    def _fetch_asset(name, asset_hash, algorithm, locations, expire, queue):
+        cache_dirs = settings.as_dict().get('datadir.paths.cache_dirs')
```

Comment:
This brings an interesting question: given the fact that a subset of the configuration is already passed to the runner, shouldn't it be used here?

Also, AFAICT this reading from the settings modules does not guarantee that a the configuration on a job/suite would be reflected here.

Now, the challenge is how to pick and choose the relevant subset of settings to the runner.

Reply:
> This brings an interesting question: given the fact that a subset of the configuration is already passed to the runner, shouldn't it be used here?

Yes! I based the development on https://github.com/avocado-framework/avocado/blob/a3667c7ea1f4d92889a37353e1da0e4601e9c48e/avocado/plugins/assets.py#L198 and forgot we may already have the information needed in the runner.

> 
> Also, AFAICT this reading from the settings modules does not guarantee that a the configuration on a job/suite would be reflected here.

Indeed.

> 
> Now, the challenge is how to pick and choose the relevant subset of settings to the runner.

Let's see if I can come up with a solution :)